### PR TITLE
feat: module docs

### PR DESCRIPTION
**Pull request by fgdorais**
Add statement of purpose as requested in #203.

### CODE REVIEW

File: Std/Data/Nat/Lemmas.lean [hunk 1]
```diff
@@ -344,7 +351,7 @@ protected theorem add_self_ne_one : ∀ (n : Nat), n + n ≠ 1
     have h1 : succ (succ (n + n)) = 1 := succ_add n n ▸ h
     Nat.noConfusion h1 fun.
 
-/-! ### sub -/
+/-! ## sub -/
```
Comment (YaelDillies) [inline Std/Data/Nat/Lemmas.lean:354]: I think it makes sense to keep the headers with the same size as everywhere else even if we're not using the intermediate size.

**Review by YaelDillies — APPROVED** [review]
Thank you! I'm happy with this docstring, but I guess Mario should sign off.

File: Std/Data/Nat/Lemmas.lean (hunk 1 above)
Comment (digama0) [reply Std/Data/Nat/Lemmas.lean:354]: I don't think we are being very consistent about header size. It probably requires a global review.

Comment (YaelDillies) [reply Std/Data/Nat/Lemmas.lean:354]: At least, we're consistent within mathlib.

Comment (fgdorais) [reply Std/Data/Nat/Lemmas.lean:354]: Should I change from `##` to `###`? The file was inconsistent so I picked one. I have no personal preference for 2 or 3.

Comment (YaelDillies) [reply Std/Data/Nat/Lemmas.lean:354]: I'd personally go for `###` but I guess Mario might have a different opinion.

File: Std/Data/Nat/Lemmas.lean [hunk 2]
```diff
@@ -8,6 +8,13 @@ import Std.Tactic.Basic
 import Std.Data.Nat.Init.Lemmas
 import Std.Data.Nat.Basic
 
+/-! # Basic lemmas about natural numbers
+
+The primary purpose of the lemmas in this file is to assist with reasonning
```
Comment (semorrison) [inline Std/Data/Nat/Lemmas.lean:13]:
```suggestion
The primary purpose of the lemmas in this file is to assist with reasoning
```

**Review by semorrison — APPROVED** [review]

**Pull request merged**